collector/zfs_linux.go

```diff
@@ -81,6 +98,50 @@ func (c *zfsCollector) parseProcfsFile(reader io.Reader, fmtExt string, handler
 	return scanner.Err()
 }
 
-func (c *zfsCollector) updatePoolStats(ch chan<- prometheus.Metric) (err error) {
+func (c *zfsCollector) parsePoolProcfsFiles(zpoolPaths []string, handler func(string, zfsSysctl, int)) (err error) {
+
+	for _, zpoolPath := range zpoolPaths {
+		file, err := os.Open(zpoolPath)
+		if err != nil {
+			log.Debugf("Cannot open %q for reading. Is the kernel module loaded?", zpoolPath)
+			err = zfsNotAvailableError
+		}
+		defer file.Close()
+
+		scanner := bufio.NewScanner(file)
+
+		parseLine := false
+		var fields []string
+		for scanner.Scan() {
+
+			line := strings.Fields(scanner.Text())
+
+			if !parseLine && len(line) >= 12 && line[0] == "nread" {
+				//Start parsing from here.
+				parseLine = true
+				fields = make([]string, len(line))
+				copy(fields, line)
+				continue
+			}
+			if !parseLine {
+				continue
+			}
+
+			zpoolPathElements := strings.Split(zpoolPath, "/")
```

Comment:
I see.  I think just checking for some minimum number of elements present to make the following two lines not panic in case of bad input, would be a good idea.  So maybe `if len(zpoolPathElements) < 2`, error?